tagstudio/src/core/library.py

```diff
@@ -1320,6 +1319,47 @@ def get_entry_id_from_filepath(self, filename: Path):
         except KeyError:
             return -1
 
+    def parse_metadata(
+        self, query: str | None = None
+    ) -> list[dict[str, str | list[str]]]:
+        """
+        Splits query into several maps 'meta_key -> value'\n
+        Values without specified key parsed as tags and put in 'unbound' key \n
+        example query1:
+            "meta_first: value; meta_second: value; tag1; | meta_first: value; meta_second: value; notag;"
+        example query2:
+            "tag1 | notag | tag2"
+        """
+        if query is None:
+            return []
+        meta_list: list = []
+        meta_conditions = query.strip().split("|")
+        for meta_condition in meta_conditions:
+            query_part: dict = {}
+            field_data_list = meta_condition.strip().split(";")
+            for field_data in field_data_list:
+                field_parsed = field_data.strip().split(":")
+                if len(field_parsed) < 2:
+                    unbound_values = (
+                        field_parsed[0].strip().lower().casefold().split(" ")
```

Comment:
`"two  spaces  inbetween".split(" ")` will produce `['two', '', 'spaces', '', 'inbetween']`
`.split()` would be better, as it will produce the expected `['two', 'spaces', 'inbetween']`
```suggestion
                        field_parsed[0].strip().lower().casefold().split()
```